internal/tests/cluster/x509_verification_test.go

```diff
@@ -185,13 +186,22 @@ func TestCustomX509Verification_Client(t *testing.T) {
 }
 
 func TestCustomX509Verification_Server(t *testing.T) {
+	t.Skip("These tests are currently not working and will need further investigation")
```

Comment:
Yeah I raised and issue: https://hashicorp.atlassian.net/browse/ICU-15052. I think they were broken by 1.23 and something in the TLS stack, I had a quick look and couldn't see anything obvious in the release notes but it would be nice to fix and reenable them.